google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/models/FiltersTest.java

```diff
@@ -528,4 +538,66 @@ public void labelTest() {
 
     assertThat(actualFilter).isEqualTo(expectedFilter);
   }
+
+  @Test
+  public void serializationTest() throws InvocationTargetException, IllegalAccessException {
+    // checks that the all objects returned by the all methods of the Filters class
+    // can be serialized/deserialized.
+
+    for (Method m : Filters.class.getDeclaredMethods()) {
+      String name = m.getName();
+      if (Modifier.isPublic(m.getModifiers())) {
+        switch (name) {
+          case "condition":
+            checkSerialization(
+                name,
+                FILTERS
+                    .condition(
+                        FILTERS
+                            .chain()
+                            .filter(FILTERS.qualifier().exactMatch("data_plan_10gb"))
+                            .filter(FILTERS.value().exactMatch("true")))
+                    .then(FILTERS.label("passed-filter"))
+                    .otherwise(FILTERS.label("filtered-out")));
+            break;
+          case "label":
+            checkSerialization(name, FILTERS.label("label"));
+            break;
+          case "fromProto":
+            checkSerialization(name, FILTERS.label("label").toProto());
+            break;
+          default:
+            checkSerialization(name, m.invoke(FILTERS));
+        }
+      }
+    }
+  }
+
+  private void checkSerialization(String name, Object filter) {
+    try {
+      Object deserialized = serializeDeserialize(filter);
+      if (filter instanceof Filters.Filter) {
+        RowFilter protoBefore = ((Filters.Filter) filter).toProto();
+        RowFilter protoAfter = ((Filters.Filter) deserialized).toProto();
+        assertEquals(
+            "'" + name + "' filter protoBuf mismatch after deserialization",
+            protoBefore,
+            protoAfter);
+      }
+    } catch (IOException | ClassNotFoundException e) {
+      fail(name + ": " + e);
+    }
+  }
+
+  private Object serializeDeserialize(Object obj) throws IOException, ClassNotFoundException {
```

Comment:
static?

Reply:
yes. 
I'm thinking about moving this method to a utility class because it's very generic and could be used by any tests for serialization. Do you know a better place for it?

Reply:
done